src/cargo/core/compiler/output_sbom.rs

```diff
@@ -0,0 +1,292 @@
+//! cargo-sbom precursor files for external tools to create SBOM files from.
+//! See [`output_sbom`] for more.
+
+use std::collections::BTreeSet;
+use std::io::BufWriter;
+use std::path::PathBuf;
+
+use cargo_util::paths::{self};
+use cargo_util_schemas::core::PackageIdSpec;
+use itertools::Itertools;
+use semver::Version;
+use serde::Serialize;
+
+use crate::core::profiles::{DebugInfo, Lto, Profile};
+use crate::core::{Target, TargetKind};
+use crate::util::Rustc;
+use crate::CargoResult;
+
+use super::{unit_graph::UnitDep, BuildRunner, CrateType, Unit};
+
+#[derive(Serialize, Clone, Debug, Copy)]
+#[serde(rename_all = "kebab-case")]
+enum SbomBuildType {
+    /// A package dependency
+    Normal,
+    /// A build script dependency
+    Build,
+}
+
+/// Typed version of a SBOM format version number.
+pub struct SbomFormatVersion<const V: u32>;
+
+impl<const V: u32> Serialize for SbomFormatVersion<V> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_u32(V)
+    }
+}
+
+/// A profile can be overriden for individual packages.
+///
+/// This wraps a [`Profile`] object.
+/// See <https://doc.rust-lang.org/nightly/cargo/reference/profiles.html#overrides>
+#[derive(Serialize, Clone, Debug)]
+struct SbomProfile {
```

Comment:
No case is put on this.  Is `snake_case` intentional? Looks like thats what we use for `cargo metadata`